src/core/ddsi/src/q_entity.c

```diff
@@ -3894,6 +4209,132 @@ static void add_proxy_builtin_endpoints(
 #undef LTE
 }
 
+#ifdef DDSI_INCLUDE_SECURITY
+
+void handshake_end_cb
+(
+  struct ddsi_handshake *handshake,
+  const struct ddsi_guid *lpguid,
+  const struct ddsi_guid *ppguid,
+  nn_wctime_t timestamp,
+  enum ddsi_handshake_state result)
+{
+  struct proxy_participant *proxypp;
+  struct participant *pp;
+  int64_t shared_secret;
+  int64_t permissions_hdl;
+  struct q_globals *gv;
+
+  DDSRT_UNUSED_ARG(timestamp);
+
+  assert(handshake);
+  assert(lpguid);
+  assert(ppguid);
+
+  gv = ddsi_handshake_get_globals(handshake);
+  assert(gv);
+
+  proxypp = ephash_lookup_proxy_participant_guid (gv->guid_hash, ppguid);
```

Comment:
I'd prefer early outs on lookup failure: it is not doing anything in those cases and it means less intendentation and less state to (mentally) track.